java/src/main/resources/com/ncc/aif/aida_ontology.shacl

```diff
@@ -60,11 +60,9 @@ aida:EntityShape
             SELECT $this
             WHERE {
                 FILTER ( NOT EXISTS {
-                        ?typeAssertion a rdf:Statement .
-                        ?typeAssertion rdf:predicate rdf:type .
-                        ?typeAssertion rdf:subject $this .
-                    } && ( NOT EXISTS { ?membership aida:clusterMember $this . } ||
-                            EXISTS { ?cluster aida:prototype $this . } )
+                        $this ^rdf:subject / rdf:predicate rdf:type .
+                    } && ( NOT EXISTS { $this ^aida:clusterMember ?membership . } ||
+                           EXISTS { $this ^aida:prototype ?cluster . } )
```

Comment:
It doesn't have to happen, but should this sparql restriction be made into a common component that both the EntityShape and EventRelationShape share?

Reply:
It's your query-- if you feel better about using a common component, then that's fine with me.  I will want to re-test at least one KB for procedural reasons, but that doesn't take long.

Reply:
Do you know how to implement this as a shared component? If not, I'm good to leave it the way it is.

Reply:
Let's leave it the way it is then.